### PR TITLE
on otp-21+ use uri_string to parse uri

### PR DESCRIPTION
**Pull request by tsloughter**
This also fixes shit because currently it seems to always be returning `undefined` for scheme and other properties. 

I'd also like to extend the request to include the rest of the information `uri_string:parse` can give -- like `userinfo`.

We should re-evaluate what OTP versions to retain support for.

### CODE REVIEW

**Comment by yurrriq** [comment]
lgtm except for https://travis-ci.org/github/elli-lib/elli/jobs/743673725. will plan to check on this tomorrow too

**Comment by yurrriq** [comment]
Won't it always be the case that `{undefined, undefined, undefined} = {Scheme, Host, Port}` for `{abs_path, FullPath}`? :+1: to this PR still, and for adding stuff like `userinfo` and `query` to the request.

**Comment by yurrriq** [comment]
Looks like `uri_string:dissect_query/1` is failing on `name=knut%3D&foo` with `{error,missing_value,<<>>}`

**Comment by yurrriq** [comment]
Seems like RFC 3986 doesn't specify, but that `uri_string:dissect_query/1` doesn't support keys without values, at least on 21.3.

**Comment by yurrriq** [comment]
aaand. you already patched it :smile:  https://github.com/erlang/otp/pull/1840

**Comment by tsloughter** [comment]
Hah, I don't remember patching that :+1: 

Did you notice what is the earliest OTP with that patch? We should probably only use `uri_string` from that version and above.

**Comment by yurrriq** [comment]
> Did you notice what is the earliest OTP with that patch?

Looks like 22.0

> We should probably only use `uri_string` from that version and above.

:+1:

File: src/elli_http.erl [hunk 1]
```diff
@@ -692,6 +692,20 @@ content_length(Headers, Body)->
 %% PATH HELPERS
 %%
 
+-ifdef(OTP_RELEASE).
```
Comment (yurrriq) [inline src/elli_http.erl:743]: `?OTP_RELEASE` was added in OTP 21, but we'll want to use `uri_string:parse/1` only for OTP >= 22.

**Review by yurrriq — CHANGES_REQUESTED** [review]

File: src/elli_http.erl (hunk 1 above)
Comment (tsloughter) [reply src/elli_http.erl:743]: Ah, damn, ok.

**Review by yurrriq — APPROVED** [review]

**Pull request merged**